src/MemoizedCombinedRegexp.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Strings;
+
+/**
+ * `CombinedRegexp` optimizes matching of multiple regular expressions.
+ * Read more about the concept in
+ * {@see https://nikic.github.io/2014/02/18/Fast-request-routing-using-regular-expressions.html}.
+ */
+final class MemoizedCombinedRegexp
```

Comment:
Does it make sense to create an interface `CombinedRegexpInterface`?

Reply:
Created abstract class instead